source/guide_puma.rst

```diff
@@ -0,0 +1,197 @@
+.. author:: Theodor Müller <info@trilliarden.net>
+.. highlight:: console
+
+.. sidebar:: Logo
+  
+  .. image:: _static/images/puma.png
+      :align: center
+
+#######
+Puma
+#######
+
+Puma_ is a Ruby_ web server built for speed and parallelism. It is designed for running Rack_ apps only. It is licensed under the BSD 3-Clause license. This guide explains how to install Puma and run a minimal, custom Ruby application.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of 
+
+  * Ruby_
+  * supervisord_
+
+Installation
+============
+
+Use ``gem`` to install the latest version of Puma:
+
+::
+
+  [isabell@stardust ~]$ gem install puma
+  Fetching: puma-3.12.0.gem (100%)
+  Building native extensions. This could take a while...
+  Successfully installed puma-3.12.0
+  1 gem installed
+  [isabell@stardust ~]$ 
+  
+Create a directory, and inside the directory a file ``rubyapp.ru`` for the application:
+
+::
+
+  [isabell@stardust ~]$ mkdir ~/rubyapp
+  [isabell@stardust ~]$ touch ~/rubyapp/rubyapp.ru
+  [isabell@stardust ~]$ 
+
+Update the permissions so that the application file is executable:
+
+::
+
+  [isabell@stardust ~]$ chmod +x ~/rubyapp/rubyapp.ru 
+  [isabell@stardust ~]$ 
+  
+This file is the Rackup file for the application. Add the following content to ``~/rubyapp/rubyapp.ru``, a simple Hello World application:
+
+.. code-block:: none
+  class HelloWorld
+    def call(env)
+      [200, {"Content-Type" => "text/plain"}, ["Hello World"]]
+    end
+  end
+
+  run HelloWorld.new
+
+
+Configuration
+=============
+
+Configure port
+--------------
+
+You need to find a free port to bind Puma to it.
+
+.. include:: includes/generate-port.rst
+
+Puma configuration file
+-----------------------
+
+Next, create a Puma configuration file, update the permissions as before,
+
+::
+
+  [isabell@stardust ~]$ touch ~/rubyapp/config.rb
+  [isabell@stardust ~]$ chmod +x ~/rubyapp/config.rb
+  [isabell@stardust ~]$
+
+and add the following content:
+
```

Comment:
There is an example code block in [the guide](https://github.com/Uberspace/lab/blob/master/STYLE.md#rules-for-guides). Can you please update this accordingly? First the general snippet, then the values according to your example. Users will just copy paste your code otherwise without changing values.

```
Create ``~/etc/services.d/ghost.ini`` with the following content:

.. warning:: Replace ``<username>`` with your username!

.. code-block:: ini

 [program:ghost]
 directory=/home/<username>/ghost
 command=env NODE_ENV=production /bin/node current/index.js

In our example this would be:

.. code-block:: ini

 [program:ghost]
 directory=/home/isabell/ghost
 command=env NODE_ENV=production /bin/node current/index.js
```

Reply:
Explicitly mention what needs to be replaced (the port)